mis_builder/models/mis_builder.py

```diff
@@ -0,0 +1,834 @@
+            d = parser.parse(c.report_instance_id.pivot_date)
+            if c.type == 'd':
+                date_from = d + timedelta(days=c.offset)
+                date_to = date_from + timedelta(days=c.duration - 1)
+                date_from = date_from.strftime(
+                    tools.DEFAULT_SERVER_DATE_FORMAT)
+                date_to = date_to.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)
+                period_ids = None
+            elif c.type == 'w':
+                date_from = d - timedelta(d.weekday())
+                date_from = date_from + timedelta(days=c.offset * 7)
+                date_to = date_from + timedelta(days=(7 * c.duration) - 1)
+                date_from = date_from.strftime(
+                    tools.DEFAULT_SERVER_DATE_FORMAT)
+                date_to = date_to.strftime(tools.DEFAULT_SERVER_DATE_FORMAT)
+                period_ids = None
+            elif c.type == 'fp':
+                period_obj = self.pool['account.period']
+                all_period_ids = period_obj.search(
+                    cr, uid,
+                    [('special', '=', False),
+                     '|', ('company_id', '=', False),
+                     ('company_id', '=', c.company_id.id)],
+                    order='date_start',
+                    context=context)
+                current_period_ids = period_obj.search(
+                    cr, uid,
+                    [('special', '=', False),
+                     ('date_start', '<=', d),
+                     ('date_stop', '>=', d),
+                     '|', ('company_id', '=', False),
+                     ('company_id', '=', c.company_id.id)],
+                    context=context)
+                if not current_period_ids:
+                    raise orm.except_orm(_("Error!"),
+                                         _("No current fiscal period for %s")
+                                         % d)
+                p = all_period_ids.index(current_period_ids[0]) + c.offset
+                if p < 0 or p >= len(all_period_ids):
+                    raise orm.except_orm(_("Error!"),
+                                         _("No such fiscal period for %s "
+                                           "with offset %d") % (d, c.offset))
+                period_ids = all_period_ids[p:p + c.duration]
+                periods = period_obj.browse(cr, uid, period_ids,
+                                            context=context)
+                date_from = periods[0].date_start
+                date_to = periods[-1].date_stop
+            else:
+                raise orm.except_orm(_("Error!"),
+                                     _("Unimplemented period type %s") %
+                                     (c.type,))
+            res[c.id] = {
+                'date_from': date_from,
+                'date_to': date_to,
+                'period_from': period_ids and period_ids[0] or False,
+                'period_to': period_ids and period_ids[-1] or False,
```

Comment:
Because ```[] and 'something'``` returns [] and I find it safer to have a real bool for a missing period.